### DISCUSSION TITLE
Blog 8

### DISCUSSION THREAD

**Discussion by ktoffel999** [Anastylos/Jump-and-Run#84] [Blog]
Diese Woche haben wir eine CI/CD-Pipeline in GitHub erstellt (siehe .github/workflows/build.yml)
<img width="821" alt="Bildschirmfoto 2024-12-03 um 09 25 10" src="https://github.com/user-attachments/assets/edaba47a-d786-41f9-b4f4-06b53065de5d">

Struktur der Pipeline
	•	Trigger (on):
	•	Die Pipeline wird ausgelöst bei:
	•	push-Events auf den main-Branch.
	•	pull_request-Events auf den main-Branch.

Jobs
1. Build-Job (build):
	•	Läuft auf: ubuntu-latest.
Schritte:
	1.	Repository klonen:
	•	Nutzt den actions/checkout@v3-Action, um das aktuelle Repository zu klonen.
	2.	Godot herunterladen und installieren:
	•	Lädt die Godot-Engine (Version 4.3) als Zip-Datei von GitHub herunter.
	•	Entpackt das Zip (unzip godot.zip).
	•	Listet die Inhalte des Verzeichnisses (ls godot), um sicherzustellen, dass die Datei korrekt heruntergeladen wurde.
	•	Setzt die Datei als ausführbar (chmod +x), damit Godot ohne GUI verwendet werden kann.
	3.	Projekt validieren:
	•	Führt Godot im Headless-Modus aus, um das Projekt zu prüfen.
	•	Der Befehl --check-syntax überprüft, ob das Projekt fehlerfrei ist.
	4.	Build für Plattformen erstellen:
	•	Erstellt zwei Builds:
	•	Linux-Build: Exportiert ein ausführbares Linux-Programm (build/MyGame.x86_64).
	•	Windows-Build: Exportiert eine .exe-Datei für Windows (build/MyGame.exe).

2. Deploy-Job (deploy):
	•	Abhängigkeit: Dieser Job startet erst, wenn der build-Job erfolgreich abgeschlossen wurde (needs: build).
Schritte:
	1.	Artifact hochladen:
	•	Nutzt die softprops/action-gh-release@v1-Action, um die erzeugten Dateien (Linux und Windows Builds) als Release-
                Assets auf GitHub hochzuladen.
	•	Die Dateien sind:
	•	build/MyGame.x86_64
	•	build/MyGame.exe

**Comment by j1tbk** [comment]
Sehr übersichtlich und verständlich erklärt. Eine gelungene Pipeline mit klarer Struktur. 10/10, Daumen hoch.
Euer Team Health.

**Comment by carinabsn** [comment]
Eure Pipeline ist hervorragend strukturiert und nutzt GitHub Actions effektiv. Besonders die automatische Validierung mit Godot und die plattformübergreifenden Builds sind beeindruckend. Klasse Arbeit!

Euer GymTrackerX
